linter/lints/crl/lints.go

```diff
@@ -386,120 +391,161 @@ func hasMozReasonCodes(crl *crl_x509.RevocationList) *lint.LintResult {
-		return &parseFail
+		return &lint.LintResult{Status: lint.Error, Details: "Failed to read thisUpdate timestamp"}
 	}
+
 	// Lint thisUpdate.
 	err := lintTimestamp(&thisUpdate, thisUpdateTag)
 	if err != nil {
-		parseFail.Details = err.Error()
-		return &parseFail
+		return &lint.LintResult{Status: lint.Error, Details: err.Error()}
 	}
+
 	// Peek (optional) nextUpdate.
 	if tbs.PeekASN1Tag(cryptobyte_asn1.UTCTime) || tbs.PeekASN1Tag(cryptobyte_asn1.GeneralizedTime) {
+
 		// Read nextUpdate.
 		var nextUpdate cryptobyte.String
 		var nextUpdateTag cryptobyte_asn1.Tag
 		if !tbs.ReadAnyASN1Element(&nextUpdate, &nextUpdateTag) {
-			return &parseFail
+			return &lint.LintResult{Status: lint.Error, Details: "Failed to read nextUpdate timestamp"}
 		}
+
 		// Lint nextUpdate.
 		err = lintTimestamp(&nextUpdate, nextUpdateTag)
 		if err != nil {
-			parseFail.Details = err.Error()
-			return &parseFail
+			return &lint.LintResult{Status: lint.Error, Details: err.Error()}
 		}
 	}
-	// Read sequence of revokedCertificate.
-	var revokedSeq cryptobyte.String
-	tbs.ReadASN1(&revokedSeq, cryptobyte_asn1.SEQUENCE)
-	for !revokedSeq.Empty() {
-		// Read revokedCertificate.
-		var certSeq cryptobyte.String
-		revokedSeq.ReadASN1Element(&certSeq, cryptobyte_asn1.SEQUENCE)
-		certSeq.ReadASN1(&certSeq, cryptobyte_asn1.SEQUENCE)
 
-		// Skip userCertificate (serial number).
-		certSeq.SkipASN1(cryptobyte_asn1.INTEGER)
-
-		// Read revocationDate.
-		var revocationDate cryptobyte.String
-		var revocationDateTag cryptobyte_asn1.Tag
-		if !certSeq.ReadAnyASN1Element(&revocationDate, &revocationDateTag) {
-			return &parseFail
+	// Peek (optional) revokedCertificates.
+	if tbs.PeekASN1Tag(cryptobyte_asn1.SEQUENCE) {
+		// Read sequence of revokedCertificate.
+		var revokedSeq cryptobyte.String
+		if !tbs.ReadASN1(&revokedSeq, cryptobyte_asn1.SEQUENCE) {
+			return &lint.LintResult{Status: lint.Error, Details: "Failed to read revokedCertificate sequence"}
 		}
 
-		// Lint revocationDate.
-		err = lintTimestamp(&revocationDate, revocationDateTag)
-		if err != nil {
-			parseFail.Details = err.Error()
-			return &parseFail
+		// Iterate over each revokedCertificate sequence.
+		for !revokedSeq.Empty() {
+			// Read revokedCertificate.
+			var certSeq cryptobyte.String
+			if !revokedSeq.ReadASN1Element(&certSeq, cryptobyte_asn1.SEQUENCE) {
+				return &lint.LintResult{Status: lint.Error, Details: "Failed to read revokedCertificate in sequence"}
+			}
+
+			if !certSeq.ReadASN1(&certSeq, cryptobyte_asn1.SEQUENCE) {
+				return &lint.LintResult{Status: lint.Error, Details: "Failed to read revokedCertificate in sequence"}
+			}
+
+			// Skip userCertificate (serial number).
+			if !certSeq.SkipASN1(cryptobyte_asn1.INTEGER) {
+				return &lintFail
+			}
+
+			// Read revocationDate.
+			var revocationDate cryptobyte.String
+			var revocationDateTag cryptobyte_asn1.Tag
+			if !certSeq.ReadAnyASN1Element(&revocationDate, &revocationDateTag) {
+				return &lintFail
```

Comment:
(See comment above) Especially if this one (which is exactly the same as the ReadAnyASN1Element calls on lines 432 and 448) is not going to return a specialized error message, it's unclear to the reader why the ones above return a specialized error message.